# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/workers/WorkerScope.cpp
# Commit: cbe7670b4ae5
# Full Hash: cbe7670b4ae5f062939d18ae24af9cc2d7a8da23
# Author: Eden Chuang <echuang@mozilla.com>
# Date: 2022-01-25 16:37:35
# Regressor Bug: 1744025
# File Overlap Count: 3
# Description:
#   Bug 1744025: Use CheckedUnsafePtr<WorkerPrivate> in WorkerGlobalScopeBase. r=dom-worker-reviewers,asuth
#   
#   Depends on D132708
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D132709
# ==============================================================================

diff -r dfb7e4b76f92 -r cbe7670b4ae5 dom/workers/WorkerScope.cpp
--- a/dom/workers/WorkerScope.cpp	Tue Jan 25 08:52:59 2022 +0000
+++ b/dom/workers/WorkerScope.cpp	Tue Jan 25 08:53:00 2022 +0000
@@ -218,8 +218,7 @@
 NS_INTERFACE_MAP_END_INHERITING(DOMEventTargetHelper)
 
 WorkerGlobalScopeBase::WorkerGlobalScopeBase(
-    NotNull<WorkerPrivate*> aWorkerPrivate,
-    UniquePtr<ClientSource> aClientSource)
+    WorkerPrivate* aWorkerPrivate, UniquePtr<ClientSource> aClientSource)
     : mWorkerPrivate(aWorkerPrivate),
       mClientSource(std::move(aClientSource)),
       mSerialEventTarget(aWorkerPrivate->HybridEventTarget()) {
@@ -781,9 +780,9 @@
                                                WorkerGlobalScope)
 
 DedicatedWorkerGlobalScope::DedicatedWorkerGlobalScope(
-    NotNull<WorkerPrivate*> aWorkerPrivate,
-    UniquePtr<ClientSource> aClientSource, const nsString& aName)
-    : WorkerGlobalScope(aWorkerPrivate, std::move(aClientSource)),
+    WorkerPrivate* aWorkerPrivate, UniquePtr<ClientSource> aClientSource,
+    const nsString& aName)
+    : WorkerGlobalScope(std::move(aWorkerPrivate), std::move(aClientSource)),
       NamedWorkerGlobalScopeMixin(aName) {}
 
 bool DedicatedWorkerGlobalScope::WrapGlobalObject(
@@ -955,9 +954,9 @@
 }
 
 SharedWorkerGlobalScope::SharedWorkerGlobalScope(
-    NotNull<WorkerPrivate*> aWorkerPrivate,
-    UniquePtr<ClientSource> aClientSource, const nsString& aName)
-    : WorkerGlobalScope(aWorkerPrivate, std::move(aClientSource)),
+    WorkerPrivate* aWorkerPrivate, UniquePtr<ClientSource> aClientSource,
+    const nsString& aName)
+    : WorkerGlobalScope(std::move(aWorkerPrivate), std::move(aClientSource)),
       NamedWorkerGlobalScopeMixin(aName) {}
 
 bool SharedWorkerGlobalScope::WrapGlobalObject(
@@ -989,10 +988,9 @@
 NS_IMPL_RELEASE_INHERITED(ServiceWorkerGlobalScope, WorkerGlobalScope)
 
 ServiceWorkerGlobalScope::ServiceWorkerGlobalScope(
-    NotNull<WorkerPrivate*> aWorkerPrivate,
-    UniquePtr<ClientSource> aClientSource,
+    WorkerPrivate* aWorkerPrivate, UniquePtr<ClientSource> aClientSource,
     const ServiceWorkerRegistrationDescriptor& aRegistrationDescriptor)
-    : WorkerGlobalScope(aWorkerPrivate, std::move(aClientSource)),
+    : WorkerGlobalScope(std::move(aWorkerPrivate), std::move(aClientSource)),
       mScope(NS_ConvertUTF8toUTF16(aRegistrationDescriptor.Scope()))
 
       // Eagerly create the registration because we will need to receive